src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -257,13 +254,11 @@ private void _ReadResources()
 
             // Read RuntimeResourceSet header
             // Do file version check
-            int version = _store.ReadInt32();
-            if (version != ResSetVersion && version != 1)
-                throw new ArgumentException(SR.Arg_ResourceFileUnsupportedVersion + "Expected:" + ResSetVersion + "but got:" + version);
-            _version = version;
+            if (!_stream.TryReadInt32(out _version) || _version != ResSetVersion && _version != 1)
+                throw new ArgumentException(SR.Arg_ResourceFileUnsupportedVersion + "Expected:" + ResSetVersion + "but got:" + _version);
```

Comment:
Preexisting your change, but this exception message needs to be localizable and not contain hardcoded English strings.